client/relay-chain-minimal-node/src/network.rs

```diff
@@ -0,0 +1,420 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use futures::FutureExt;
+use polkadot_core_primitives::{Block, Hash};
+use polkadot_service::{BlockT, HeaderBackend, NumberFor};
+
+use polkadot_node_network_protocol::PeerId;
+use sc_network::{NetworkService, SyncState};
+
+use sc_network_common::sync::{Metrics, SyncStatus};
+use sc_network_light::light_client_requests;
+use sc_network_sync::{block_request_handler, state_request_handler};
+use sc_service::{error::Error, Configuration, NetworkStarter, SpawnTaskHandle};
+use sp_consensus::BlockOrigin;
+use sp_runtime::Justifications;
+
+use std::sync::Arc;
+
+use crate::BlockChainRpcClient;
+
+pub(crate) struct BuildCollatorNetworkParams<'a> {
+	/// The service configuration.
+	pub config: &'a Configuration,
+	/// A shared client returned by `new_full_parts`.
+	pub client: Arc<BlockChainRpcClient>,
+	/// A handle for spawning tasks.
+	pub spawn_handle: SpawnTaskHandle,
+	/// Genesis hash
+	pub genesis_hash: Hash,
+}
+
+struct DummyImportQueue;
```

Comment:
Could also be moved down to the end of the file to have a clean separation between the dummy stuff and the real implementation.